include/rapidjson/internal/dtoa.h

```diff
@@ -102,7 +102,8 @@ inline void DigitGen(const DiyFp& W, const DiyFp& Mp, uint64_t delta, char* buff
         kappa--;
         if (p2 < delta) {
             *K += kappa;
-            GrisuRound(buffer, *len, delta, p2, one.f, wp_w.f * kPow10[-static_cast<int>(kappa)]);
+            int index = -static_cast<int>(kappa);
+            GrisuRound(buffer, *len, delta, p2, one.f, wp_w.f * (index < 9 ? kPow10[-static_cast<int>(kappa)] : 0));
```

Comment:
Was this cause array out of bound?
Otherwise should change to `size_t` and `assert(kappa <= 0)`?

Reply:
I want to ask a question here. Since the array kPow10 is of length 10, why not check "index < 10" here? If index is 9, then it is still safe to access kPow10[9]. @efidler